doc/glossary.rst

```diff
@@ -0,0 +1,433 @@
+
+    outlier detector
+        TODO
+
+    predictor
+        An :term:`estimator` which provides :term:`predict`.
+        This encompasses :term:`classifier`, :term:`regressor`,
+        :term:`outlier detector` and sometimes :term:`clusterer` (at least when
+        they are inductive).  In scikit-learn, if an estimator is not a
+        predictor, it is usually a :term:`transformer`.
+
+    regressor
+        TODO
+        Mention :func:`~base.is_regressor`.
+
+    transformer
+        TODO
+
+    vectorizer
+        A :term:`tranformer` which takes input where each sample is not
+        represented as an :term:`array-like` object of fixed length, and
+        produces an `array-like` object for each sample (and thus a
+        2-dimensional array-like for a set of samples).
+
+There are further APIs specific to a small family of estimators, such as:
+
+* :class:`neighbors.DistanceMetric`
+* :class:`gaussian_process.kernels.Kernel`
+* ``tree.Criterion``
+
+Target Types
+============
+
+.. glossary::
+
+    binary
+        TODO
+
+    continuous
+        TODO
+
+    multiclass
+        TODO
+
+    multilabel
+        TODO
+
+    multi-output continuous
+        TODO
+
+    multi-output multiclass
+        TODO
+
+Methods
+=======
+
+.. glossary::
+
+    ``decision_function``
+
+        classifier
+            TODO
+
+        outlier detector
+            TODO
+
+    ``get_feature_names``
+        TODO
+
+    ``get_n_splits``
+        TODO
+
+    ``get_params``
+        TODO
+
+    ``fit_predict``
+        TODO
+
+    ``fit_transform``
+        TODO
+
+    ``fit``
+        The ``fit`` method is provided on every estimator. It usually takes some
+        :term`samples` ``X``, :term:`targets` ``y`` if the model is supervised,
+        and potentially other :term:`sample properties` such as
+        :term:`sample_weight`.  It should:
+
+        * clear any prior :term:`attributes` stored on the estimator, unless
+          :term:`warm_start` is used;
+        * validate and interpret any :term:`parameters`, ideally raising an
+          error if invalid;
+        * validate the input data;
+        * estimate and store model attributes from the estimated parameters and
+          provided data; and
+        * return the now :term:`fitted` estimator to facilitate method
+          chaining.
+
+    ``partial_fit``
+        TODO
+
+    ``predict_log_proba``
+        TODO
+
+    ``predict_proba``
+        TODO
+
+    ``predict``
+        TODO
+        Mention ``return_std``
+
+    ``score``
+        TODO
+
+    ``score_samples``
+        TODO
+
+    ``set_params``
+        TODO
+
+    ``split``
+        TODO
+
+Parameters
+==========
+
+These common parameter names, specifically used in estimator construction
+(see concept :term:`parameter`) sometimes also appear as function and
+non-estimator parameters with similar semantics.
+
+.. glossary::
+
+    ``class_weight``
+        TODO
+
+    ``cv``
+        TODO
+
+    ``max_iter``
```

Comment:
I'm trying to document this. Does someone else have a sense of when this means "epochs" == "passes over the full X" and when it means something else (if ever)?